src/overview/container.js

```diff
@@ -214,15 +276,28 @@ class OverviewContainer extends Component {
             !wereAnyClicked(this.furtherTagRefs)
         ) {
             this.props.resetActiveTagIndex()
+            this.props.resetFilterPopup()
         }
     }
 
     render() {
+        const { filterPopup, filterTags, filterDomains } = this.props
+
         return (
             <Overview
                 {...this.props}
                 setInputRef={this.setInputRef}
                 onInputChange={this.props.handleInputChange}
+                tagFilterManager={this.renderTagsFilter(filterPopup === 'tag')}
```

Comment:
these state derivations need to be moved to selectors. Just replace the `filterPopup` selector with selectors for those `shouldDisplayDomainFilterPopup` and `shouldDisplayTagFilterPopup` bools you are passing into the render methods. They also won't need to pass around anything, just check `this.props.x`